src/cpp/src/image_generation/stable_diffusion_3_pipeline.hpp

```diff
@@ -273,25 +273,36 @@ class StableDiffusion3Pipeline : public DiffusionPipeline {
         std::string negative_prompt_3_str = generation_config.negative_prompt_3 != std::nullopt ? *generation_config.negative_prompt_3 : negative_prompt_1_str;
 
         // text_encoder_1_output - stores positive and negative pooled_prompt_embeds
+        auto infer_start = std::chrono::steady_clock::now();
         ov::Tensor text_encoder_1_output = m_clip_text_encoder_1->infer(positive_prompt, negative_prompt_1_str, do_classifier_free_guidance(generation_config.guidance_scale));
+        auto infer_duration = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::steady_clock::now() - infer_start).count();
+        m_perf_metrics.encoder_inference_duration["text_encode"] = infer_duration;
 
         // text_encoder_1_hidden_state - stores positive and negative prompt_embeds
         size_t idx_hidden_state_1 = m_clip_text_encoder_1->get_config().num_hidden_layers + 1;
         ov::Tensor text_encoder_1_hidden_state = m_clip_text_encoder_1->get_output_tensor(idx_hidden_state_1);
 
         // text_encoder_2_output - stores positive and negative pooled_prompt_2_embeds
+        infer_start = std::chrono::steady_clock::now();
         ov::Tensor text_encoder_2_output = m_clip_text_encoder_2->infer(prompt_2_str, negative_prompt_2_str, do_classifier_free_guidance(generation_config.guidance_scale));
+        infer_duration = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::steady_clock::now() - infer_start).count();
+        m_perf_metrics.encoder_inference_duration["text_encode_2"] = infer_duration;
 
         // text_encoder_2_hidden_state - stores positive and negative prompt_2_embeds
         size_t idx_hidden_state_2 = m_clip_text_encoder_2->get_config().num_hidden_layers + 1;
         ov::Tensor text_encoder_2_hidden_state = m_clip_text_encoder_2->get_output_tensor(idx_hidden_state_2);
 
         ov::Tensor text_encoder_3_output;
         if (m_t5_text_encoder) {
+            infer_start = std::chrono::steady_clock::now();
             text_encoder_3_output = m_t5_text_encoder->infer(prompt_3_str,
                                                              negative_prompt_3_str,
                                                              do_classifier_free_guidance(generation_config.guidance_scale),
                                                              generation_config.max_sequence_length);
+            auto infer_duration =
+                std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::steady_clock::now() - infer_start)
+                    .count();
+            m_perf_metrics.encoder_inference_duration["text_encode_3"] = infer_duration;
```

Comment:
maybe we need to explicitly write `0` as `text_encoder_3` time in case it's not used?